src/shims/unix/linux/epoll.rs

```diff
@@ -431,28 +431,42 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let mut num_of_events: i32 = 0;
         let mut array_iter = this.project_array_fields(&event)?;
 
-        while let Some((epoll_key, epoll_return)) = ready_list.pop_first() {
-            // If the file description is fully close, the entry for corresponding FdID in the
-            // global epoll event interest table would be empty.
-            if this.machine.epoll_interests.get_epoll_interest(epoll_key.0).is_some() {
-                // Return notification to the caller if the file description is not fully closed.
-                if let Some(des) = array_iter.next(this)? {
-                    this.write_int_fields_named(
-                        &[
-                            ("events", epoll_return.events.into()),
-                            ("u64", epoll_return.data.into()),
-                        ],
-                        &des.1,
-                    )?;
-                    num_of_events = num_of_events.checked_add(1).unwrap();
-                } else {
-                    break;
-                }
+        while let Some(des) = array_iter.next(this)? {
+            if let Some(epoll_event_instance) = this.ready_list_next(&mut ready_list) {
+                this.write_int_fields_named(
+                    &[
+                        ("events", epoll_event_instance.events.into()),
+                        ("u64", epoll_event_instance.data.into()),
+                    ],
+                    &des.1,
+                )?;
+                num_of_events = num_of_events.checked_add(1).unwrap();
+            } else {
+                break;
             }
         }
         Ok(Scalar::from_i32(num_of_events))
     }
 
+    /// This function takes in ready list and returns EpollEventInstance with file description
+    /// that is not closed.
+    fn ready_list_next(
```

Comment:
You are adding this as a public function that can be called everywhere in Miri -- that's not the intention, as this is just a private helper. Please make the function private.

Reply:
I moved this out and turned this a free function. Alternatively, we can create a wrapper type for ``ready_list`` and make this a method of that type.

Reply:
A free function sounds good for this!